src/persist-client/src/internal/metrics.rs

```diff
@@ -987,76 +988,61 @@ impl ShardsMetrics {
-            encoded_batch_size: registry.register(
-                metric!(
-                    name: "mz_persist_shard_encoded_batch_size",
-                    help: "total encoded batch size of all active shards on this process",
-                    var_labels: ["shard"],
-                ),
-            ),
-            seqnos_held: registry.register(
-                metric!(
-                    name: "mz_persist_shard_seqnos_held",
-                    help: "maximum count of gc-ineligible states across all active shards on this process",
-                    var_labels: ["shard"],
-                ),
-            ),
-            gc_finished: registry.register(
-                metric!(
-                    name: "mz_persist_shard_gc_finished",
-                    help: "count of garbage collections finished by shard",
-                    var_labels: ["shard"],
-                ),
-            ),
-            compaction_applied: registry.register(
-                metric!(
-                    name: "mz_persist_shard_compaction_applied",
-                    help: "count of compactions applied to state by shard",
-                    var_labels: ["shard"],
-                ),
-            ),
+            since: registry.register(metric!(
+                name: "mz_persist_shard_since",
+                help: "since by shard",
+                var_labels: ["shard"],
+            )),
+            upper: registry.register(metric!(
+                name: "mz_persist_shard_upper",
+                help: "upper by shard",
+                var_labels: ["shard"],
+            )),
+            encoded_rollup_size: registry.register(metric!(
+                name: "mz_persist_shard_rollup_size_bytes",
+                help: "total encoded rollup size by shard",
+                var_labels: ["shard"],
+            )),
+            encoded_diff_size: registry.register(metric!(
+                name: "mz_persist_shard_diff_size_bytes",
+                help: "total encoded diff size by shard",
+                var_labels: ["shard"],
+            )),
+            batch_part_count: registry.register(metric!(
+                name: "mz_persist_shard_batch_part_count",
+                help: "count of batch parts by shard",
+                var_labels: ["shard"],
+            )),
+            update_count: registry.register(metric!(
+                name: "mz_persist_shard_update_count",
+                help: "count of updates by shard",
+                var_labels: ["shard"],
+            )),
+            encoded_batch_size: registry.register(metric!(
+                name: "mz_persist_shard_encoded_batch_size",
+                help: "total encoded batch size by shard",
+                var_labels: ["shard"],
+            )),
+            largest_batch_size: registry.register(metric!(
+                name: "mz_persist_shard_largest_batch_size",
+                help: "largest encoded batch size by shard",
+                var_labels: ["shard"],
```

Comment:
Is this the largest batch size ever seen in the shard, or the largest one that exists right now? I think the former, in which case it might be helpful to reflect that in the help text.

Reply:
The largest one that exists right now. The largest over time could then be queried in Prometheus (at least for as long as our retention window is)

Reply:
yeah, what paul said!

Reply:
Ah, perfect!